Assignment2

```diff
@@ -0,0 +1,56 @@
+#include<iostream>
+using namespace std;
+
+__global__ void Add(float *d_a,float *d_b,float *d_c,int r,int c){
+    
+    int i =blockIdx.x*blockDim.x+threadIdx.x;
+    int j =blockIdx.y*blockDim.y+threadIdx.y;
+    int k = i+j*c;
+    if(i<r && j<c){
+        d_c[k]=d_a[k]+d_b[k];
+    }    
+}
+
+
+int main()
+{
+    int r,c,i,j;
+    cout<<"Enter the rows and columns\n";
+    cin>>r>>c;
+    float h_a[r][c],h_b[r][c],h_c[r][c];
+    for(i=0;i<r;i++)
+    {
+        for(j=0;j<c;j++)
+        {
+            h_a[i][j]=i+j+3;
+            h_b[i][j]=i*j;
+        }
+    }
+    float *d_a,*d_b,*d_c;
+    cudaMalloc((void**)&d_a, (r*c)*sizeof(float));
+    cudaMalloc((void**)&d_b, (r*c)*sizeof(float));
+    cudaMalloc((void**)&d_c, (r*c)*sizeof(float));
+
+    cudaMemcpy(d_a, h_a, r*c*sizeof(float), cudaMemcpyHostToDevice);
+    cudaMemcpy(d_b, h_b, r*c*sizeof(float), cudaMemcpyHostToDevice);
+
+    
+    dim3 threadPerBlock(r,c);
+    Add<<<1,threadPerBlock>>>(d_a,d_b,d_c,r,c);
```

Comment:
Also think of launching multiple blocks, since a block can have a maximum of 1024 threads. So your code will not work on matrices having more than 1024 elements.